crates/nu-command/src/math/mode.rs

```diff
@@ -0,0 +1,174 @@
+use crate::math::utils::run_with_function;
+use nu_protocol::ast::Call;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{Example, PipelineData, ShellError, Signature, Span, Value};
+use std::cmp::Ordering;
+
+#[derive(Clone)]
+pub struct SubCommand;
+
+#[derive(Hash, Eq, PartialEq, Debug)]
+enum NumberTypes {
+    Float,
+    Int,
+    Duration,
+    Filesize,
+}
+
+#[derive(Hash, Eq, PartialEq, Debug)]
+struct HashableType {
+    bytes: [u8; 8],
+    original_type: NumberTypes,
+}
+
+impl HashableType {
+    fn new(bytes: [u8; 8], original_type: NumberTypes) -> HashableType {
+        HashableType {
+            bytes,
+            original_type,
+        }
+    }
+}
+
+impl Command for SubCommand {
+    fn name(&self) -> &str {
+        "math mode"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("math mode")
+    }
+
+    fn usage(&self) -> &str {
+        "Gets the most frequent element(s) from a list of numbers or tables"
+    }
+
+    fn run(
+        &self,
+        _engine_state: &EngineState,
+        _stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<nu_protocol::PipelineData, nu_protocol::ShellError> {
+        run_with_function(call, input, mode)
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![Example {
+            description: "Get the mode(s) of a list of numbers",
+            example: "[3 3 9 12 12 15] | math mode",
+            result: Some(Value::List {
+                vals: vec![Value::test_int(3), Value::test_int(12)],
+                span: Span::unknown(),
+            }),
+        }]
+    }
+}
+
+pub fn mode(values: &[Value], head: &Span) -> Result<Value, ShellError> {
+    if let Some(Err(values)) = values
+        .windows(2)
+        .map(|elem| {
+            if elem[0].partial_cmp(&elem[1]).is_none() {
+                return Err(ShellError::OperatorMismatch {
+                    op_span: *head,
+                    lhs_ty: elem[0].get_type(),
+                    lhs_span: elem[0].span()?,
+                    rhs_ty: elem[1].get_type(),
+                    rhs_span: elem[1].span()?,
+                });
+            }
+            Ok(elem[0].partial_cmp(&elem[1]).unwrap())
+        })
+        .find(|elem| elem.is_err())
+    {
+        return Err(values);
+    }
+    //In e-q, Value doesn't implement Hash or Eq, so we have to get the values inside
+    // But f64 doesn't implement Hash, so we get the binary representation to use as
+    // key in the HashMap
+    let hashable_values: Result<Vec<HashableType>, ShellError> = values
+        .iter()
+        .map(|val| match val {
+            Value::Int { val, .. } => Ok(HashableType::new(val.to_be_bytes(), NumberTypes::Int)),
```

Comment:
assuming big endian may not be good. there are some functions in nushell to figure this out.
```rust
fn int_to_endian(n: i64) -> Vec<u8> {
    if cfg!(target_endian = "little") {
        n.to_le_bytes().to_vec()
    } else {
        n.to_be_bytes().to_vec()
    }
}

fn bigint_to_endian(n: &BigInt) -> Vec<u8> {
    if cfg!(target_endian = "little") {
        n.to_bytes_le().1
    } else {
        n.to_bytes_be().1
    }
}
```

Reply:
Because of the different target architectures right? That makes sense, i'm not used to having to think about bytes, thanks, gonna wait if there is a different solution that don't involve a bytes representation, and if there isn't im going to change to your sugestion.

Reply:
We could see how hard it would be to implement Hash on Value. Would that do it?

Reply:
We would need to impl Hash and Eq(It would be relatively, we would probably hash and compare for equality just the values and ignore the spans) and in the case of `Value::Float` we would try to hash and compare the bytes representation, but, we would have problem in some corner cases like 0.2 + 0.1 not being equal to 0.3. i can do it if you want, just let me known where to put the code.

Reply:
Since we're just using this for a hash, being big endian or little endian might also not matter just because the hash just needs some bits. (it'd likely matter if we tried to use it in other ways)

@fdncred - you okay with this landing if the be/le stuff is just for the hash and nothing else?

Reply:
`ne_bytes` looks good to me